### ISSUE TITLE
Laravel 9 PHP8.1 Not reporting errors, All transactions are "unknown"

### ISSUE THREAD

**Issue by iquad** [newrelic/newrelic-php-agent#392]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description
After installed new php8.1 agent, all of transactions named as "unknown" and agent not reporting laravel errors to newrelic.
It was working fine on PHP7.4 and old agent, like 2 months ago.

I did comment out distributed_tracing part on `newrelic.ini`, but no luck.

[NOTE]: # ( Describe the problem you're encountering. )
[TIP]:  # ( Do NOT give us access or passwords to your New Relic account or API keys! )

## Expected Behavior
Was expecting to see transactions as usual, and see errors. 
[NOTE]: # ( Tell us what you expected to happen. )

### newrelic-daemon.log
```
2022/03/19 07:11:11.604714 (1105605) Info: New Relic daemon version 9.20.0.310-952b9f129ea0 [listen="@newrelic" startup=agent pid=1105605 ppid=1105595 uid=0 euid=0 gid=0 egid=0 runtime="go1.9.7" GOMAXPROCS=64 GOOS=linux GOARCH=amd64]
2022/03/19 07:11:11.699716 (1105628) Info: New Relic daemon version 9.20.0.310-952b9f129ea0 [listen="@newrelic" startup=agent pid=1105628 ppid=1 uid=0 euid=0 gid=0 egid=0 runtime="go1.9.7" GOMAXPROCS=64 GOOS=linux GOARCH=amd64]
2022/03/19 07:11:11.750977 (1105735) Info: New Relic daemon version 9.20.0.310-952b9f129ea0 [listen="@newrelic" startup=agent pid=1105735 ppid=1105628 uid=0 euid=0 gid=0 egid=0 runtime="go1.9.7" GOMAXPROCS=64 GOOS=linux GOARCH=amd64]
2022/03/19 07:11:11.751015 (1105735) Info: increased file limit to 2048
2022/03/19 07:11:11.751062 (1105735) Info: collector configuration is &{CAFile: CAPath: Proxy:}
2022/03/19 07:11:11.751351 (1105735) Info: daemon listening on @newrelic
2022/03/19 07:11:13.013819 (1105735) Info: Reporting to: https://rpm.eu.newrelic.com/accounts/[ID]/applications/[APPID]
2022/03/19 07:11:13.015051 (1105735) Info: app 'foobar' connected with run id 'foobar'
```
### php_agent.log
```
2022-03-19 07:20:01.224 +0300 (1107761 1107761) info: attempt daemon connection via '@newrelic'
2022-03-19 07:20:01.224 +0300 (1107761 1107761) info: New Relic 9.20.0.310 ("allium" - "952b9f129ea0") [daemon='@newrelic'  php='8.1.2' zts=no sapi='cli'  pid=1107761 ppid=1107758 uid=0 euid=0 gid=0 egid=0 backtrace=yes startup=agent os='Linux' rel='5.4.0-99-generic' mach='x86_64' ver='#112-Ubuntu SMP Thu Feb 3 13' node='LocalWeb']
2022-03-19 07:20:13.667 +0300 (1107831 1107831) info: attempt daemon connection via '@newrelic'
2022-03-19 07:20:13.668 +0300 (1107831 1107831) info: New Relic 9.20.0.310 ("allium" - "952b9f129ea0") [daemon='@newrelic'  php='8.1.2' zts=no sapi='cli'  pid=1107831 ppid=1107830 uid=0 euid=0 gid=0 egid=0 backtrace=yes startup=agent os='Linux' rel='5.4.0-99-generic' mach='x86_64' ver='#112-Ubuntu SMP Thu Feb 3 13' node='LocalWeb']
2022-03-19 07:21:01.921 +0300 (1107894 1107894) info: attempt daemon connection via '@newrelic'
2022-03-19 07:21:01.921 +0300 (1107894 1107894) info: New Relic 9.20.0.310 ("allium" - "952b9f129ea0") [daemon='@newrelic'  php='8.1.2' zts=no sapi='cli'  pid=1107894 ppid=1107890 uid=0 euid=0 gid=0 egid=0 backtrace=yes startup=agent os='Linux' rel='5.4.0-99-generic' mach='x86_64' ver='#112-Ubuntu SMP Thu Feb 3 13' node='LocalWeb']
2022-03-19 07:22:01.996 +0300 (1108004 1108004) info: attempt daemon connection via '@newrelic'
2022-03-19 07:22:01.996 +0300 (1108004 1108004) info: New Relic 9.20.0.310 ("allium" - "952b9f129ea0") [daemon='@newrelic'  php='8.1.2' zts=no sapi='cli'  pid=1108004 ppid=1108000 uid=0 euid=0 gid=0 egid=0 backtrace=yes startup=agent os='Linux' rel='5.4.0-99-generic' mach='x86_64' ver='#112-Ubuntu SMP Thu Feb 3 13' node='LocalWeb']
2022-03-19 07:23:01.988 +0300 (1108105 1108105) info: attempt daemon connection via '@newrelic'
2022-03-19 07:23:01.988 +0300 (1108105 1108105) info: New Relic 9.20.0.310 ("allium" - "952b9f129ea0") [daemon='@newrelic'  php='8.1.2' zts=no sapi='cli'  pid=1108105 ppid=1108103 uid=0 euid=0 gid=0 egid=0 backtrace=yes startup=agent os='Linux' rel='5.4.0-99-generic' mach='x86_64' ver='#112-Ubuntu SMP Thu Feb 3 13' node='LocalWeb']
2022-03-19 07:24:01.174 +0300 (1108211 1108211) info: attempt daemon connection via '@newrelic'
2022-03-19 07:24:01.174 +0300 (1108211 1108211) info: New Relic 9.20.0.310 ("allium" - "952b9f129ea0") [daemon='@newrelic'  php='8.1.2' zts=no sapi='cli'  pid=1108211 ppid=1108207 uid=0 euid=0 gid=0 egid=0 backtrace=yes startup=agent os='Linux' rel='5.4.0-99-generic' mach='x86_64' ver='#112-Ubuntu SMP Thu Feb 3 13' node='LocalWeb']
```

[NOTE]: # ( Please provide specifics of the local error logs, Browser Dev Tools console, etc. if appropriate and possible. )

## Your Environment

[TIP]:  # ( Include as many relevant details about your environment as possible. )

* OS
```
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04
Codename:	focal
```
* PHP
```
PHP 8.1.2 (cli) (built: Jan 24 2022 10:42:33) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.1.2, Copyright (c) Zend Technologies
    with Zend OPcache v8.1.2, Copyright (c), by Zend Technologies
```

## Additional context

[TIP]:  # ( Add any other context about the problem here. )

**Comment by iMusicJJ** [comment]
We are running into the same issue;

No FPM errors, but CLI errors are reported just fine.

A temporary solution is to set `newrelic.loglevel` to `verbosedebug`.

Reducing it down to `verbose` again, prevents any errors from showing up in NR.


As with @iquad , this happened as we upgraded to PHP 8.1 and upgraded the NR Client.

**Comment by Koozza** [comment]
Ah yes! I was investigating this issue as well since we also encounter this problem. It's not only for Laravel 9 but also Laravel 8. I wasn't able to test Laravel 7 since that doesn't run on PHP 8.1

I'm a compleet NR / C beginner, so take bellow with a grain of salt but:
What I found out was that it appears to be related to opcache. Disabling opcache made all requests register correct in NR.

After compiling NR for myself and placing allot of debug lines it seems that in the cases that NR says "unknown" it just isn't receiving the call that the application is bootstrapped. When a page is succesfully parced the code will end up in the `nr_laravel_application_construct` method(?). When it's unknown it won't even end up there, and that should be the first point of entry.

I've debugged this further a few days ago but all seem to be pointing towards that the wole `Illuminate\\Foundation\\Application::__construct` just isn't being picked up by NR and is actually being used by PHP.

As mentioned above, when disabling opcache I found that it worked for 100% of the requests, while with opcache it appears to be sort of random.

**Comment by gerardnll** [comment]
I also have opcache enabled if that helps debugging.

After two weeks without an official answer I'm worried if is this being worked on...

**Comment by raulamoretti** [comment]
Laravel 8 with php 8.1 with same problem. All transactions are unknown!

**Comment by zsistla** [comment]
@Koozza , thanks for the pointers, it really helps us narrow down debugging strategies.  Please see duplicate issue https://github.com/newrelic/newrelic-php-agent/issues/387#issuecomment-1087106037 for more details.

**Comment by adduc** [comment]
After updating to 9.21.0.311, we're still seeing ~99% of transactions recorded as unknown with Laravel 9 on PHP 8.1.

**Comment by gerardnll** [comment]
Worked for a few minutes, back to 'unknown'.

**Comment by gbtagliari** [comment]
> Laravel 8 with php 8.1 with same problem. All transactions are unknown!

Here too! Can someone help?!

**Comment by zsistla** [comment]
Hi and thanks for your continued engagement.

Please see https://github.com/newrelic/newrelic-php-agent/issues/387#issuecomment-1112715244 for updates where we are continuing to track this issue.

**Comment by keithdaykin-sykes** [comment]
> > Laravel 8 with php 8.1 with same problem. All transactions are unknown!
> 
> Here too! Can someone help?!

If this is still an issue for you, I added this function to a middleware which gets called on every request:
https://docs.newrelic.com/docs/apm/agents/php-agent/php-agent-api/newrelic_name_transaction/#basic-example

**Comment by Ak-x** [comment]
This was fixed per @zsistla 's comment, in release 10.5.0+.
